build.gradle

```diff
@@ -126,7 +126,16 @@ dependencies {
     implementation 'net.jodah:expiringmap:0.5.10'
 
     // awaitility
-    testImplementation 'org.awaitility:awaitility:4.0.1'
+    implementation 'org.awaitility:awaitility:4.0.1'
+
+    // web3j dependency
```

Comment:
?

Reply:
The worker was throwing an exception when using a local `iexec-common` build (`ClassNotFoundException`) on Awaitility.

Reply:
I understand dev purposes but it should not be present here

Reply:
Alright, that's reverted. However, `iexec-core` will still depend on transitivity of `iexec-common` dependencies which IMHO is a bad idea.

Reply:
Would it suit you if we leave these lines as a comment so that we could debug faster without adding it to the deployed versions?

Reply:
I understand for transitive dependencies. We must decide what should be present and what shouldn't. For instance, our target is not having a direct web3j dependency in iExec Core Scheduler.

Reply:
Are you sure about that last point? We need it there, for example: https://github.com/iExecBlockchainComputing/iexec-core/blob/56452cea9b0db738015411dccd89e6d2ce174282/src/main/java/com/iexec/core/result/ResultService.java#L30
For reference, we currently have 5 classes directly depending on Web3J dependency in iexec-core: `DataSetService`, `IexecHubService`, `SignatureService`, `ResultService` and `WorkerController`.

But I get your point for the other dependencies as they are not directly required by iexec-core.

Reply:
Yes, if really required we might only import sub libraries:
```
// https://mvnrepository.com/artifact/org.web3j/crypto
implementation 'org.web3j:crypto:4.7.0'
```